keyboards/fancytech/fancyalice66/keymaps/via/keymap.c

```diff
@@ -0,0 +1,79 @@
+        KC_LCTL,          KC_LALT, KC_SPC,                    MO(1),   KC_SPC,                    KC_RALT,                   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT(
+        KC_TILD, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  RESET,            KC_PSCR, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, RGB_SAI, RGB_SAD, RGB_TOG, RGB_HUI,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          RGB_HUD,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          RGB_VAI,
+	    _______,          _______, _______,                   _______, _______,                   RGB_MOD,                   _______, RGB_VAD, _______
+    ),
+
+    [2] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          _______,
+	    _______,          _______, _______,                   _______, _______,                   _______,                   _______, _______, _______
+    ),
+
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          _______,
+	    _______,          _______, _______,                   _______, _______,                   _______,                   _______, _______, _______
+    )
+};
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+        tap_code(dynamic_keymap_get_keycode(biton32(layer_state), 4, 12));
+    } else {
+        tap_code(dynamic_keymap_get_keycode(biton32(layer_state), 4, 11));
+    }
+  }
+  return true;
+}
+#endif
```

Comment:
This isn't going to work well, if at all. 

Eg, this only supports basic keycodes, so no rgb keycodes, no modded keycodes, etc. 

To be honest, you want something more like this: 
https://github.com/qmk/qmk_firmware/blob/8fa7b62011484f078548535556e91494a7f6cb8d/keyboards/work_louder/encoder_actions.c

This uses the "action_exec" path, which means that EVERY keycode, including custom keycodes are supported. (though certain keycodes, like mod tap and layer tap, and the like) won't work well).

However, ideally, via should be updated to support Encoder Maps, which are keymap like arrays, that support the underlying dynamic keymap functionality that VIA relies upon.